addons/web_editor/static/src/js/widgets/widgets.js

```diff
@@ -186,18 +186,22 @@ var MediaWidget = Widget.extend({
 var ImageWidget = MediaWidget.extend({
     template: 'web_editor.dialog.image',
     events: {
-        'click .o_upload_image_button': '_onUploadButtonClick',
+        'click .o_upload_image_button, .o_upload_document_button': '_onUploadButtonClick',
         'click .o_upload_image_button_no_optimization': '_onUploadButtonNoOptimizationClick',
         'change input[type=file]': '_onImageSelection',
-        'click .o_upload_image_url_button': '_onUploadURLButtonClick',
+        'click .o_upload_image_url_button, .o_upload_document_url_button': '_onUploadURLButtonClick',
         'input input.url': "_onSearchInput",
         'click .existing-attachments [data-src]': '_onImageClick',
         'dblclick .existing-attachments [data-src]': '_onImageDblClick',
         'click .o_existing_attachment_remove': '_onRemoveClick',
+        'click .o_load_more_images': '_onLoadMore',
+        'input #imageurl': '_onImageURLInput',
+        'input #documenturl': '_onDocumentURLInput',
```

Comment:
Are not those two input event the same as the line `input input.url` above ? Should not the logic be there instead of two new functions ?